### DISCUSSION TITLE
Welcome to Deep-KAN Discussions!

### DISCUSSION THREAD

**Discussion by sidhu2690** [sidhu2690/Deep-KAN#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by rayset** [comment]
Hi there,
I've been playing around with various KAN implementations to test on MNIST, and so far yours is by far the fastest one... even faster than approximated KANs, which is honestly surprising.

I wonder if you could give some more detail about what you changed compared to the original implementation?  looking at the KAN class it seems that there is no real methodological difference, is it just a matter of implementation?

**Comment by sidhu2690** [comment]
Yes. It is faster because of its tensor-based approach for computing B-spline basis functions and optimized matrix computations provided by PyTorch. Additionally, explicit loops were avoided. Thanks for asking.

**Comment by mmghorbani** [comment]
Hello, first of all, thank you for sharing this repository. I appreciate your work. However, the comparison between MLP and KAN in the README is unfair. While you've attempted to use the same number of neurons and layers for both models, that is precisely where their fundamental differences lie. A more balanced approach would be to compare them using the same number of parameters while maintaining the same architecture.

**Comment by sidhu2690** [comment]
Thank you for your message. I am aware of this and have already conducted
experiments where I compared MLP and KAN using a similar number of
parameters. Furthermore, I have successfully applied this implementation to
solve a high-dimensional physics problem and have a comparative analysis. I
forgot to update the repository with these results. I plan to update the
repository with this information as soon as I have some free time available.

Thank you again for pointing this out.
Sidharth

**Reply by mmghorbani** [reply]
Thank you for your response, and you're welcome! It would be helpful to see the results of your experiments.
